bindings/c/test/unit/unit_tests.cpp

```diff
@@ -1743,6 +1743,191 @@ TEST_CASE("fdb_transaction_add_conflict_range") {
   CHECK(success);
 }
 
+TEST_CASE("special-key-space valid transaction ID") {
+  auto value = get_value("\xff\xff/tracing/a/transaction_id", /* snapshot */ false, {});
+  REQUIRE(value.has_value());
+  uint64_t transaction_id = std::stoul(value.value());
+  CHECK(transaction_id > 0);
+}
+
+TEST_CASE("special-key-space custom transaction ID") {
+  fdb::Transaction tr(db);
+  fdb_check(tr.set_option(FDB_TR_OPTION_SPECIAL_KEY_SPACE_ENABLE_WRITES,
+                          nullptr, 0));
+  while (1) {
+    tr.set("\xff\xff/tracing/a/transaction_id", std::to_string(ULONG_MAX));
+    fdb::ValueFuture f1 = tr.get("\xff\xff/tracing/a/transaction_id",
+                                 /* snapshot */ false);
+
+    fdb_error_t err = wait_future(f1);
+    if (err) {
+      fdb::EmptyFuture f2 = tr.on_error(err);
+      fdb_check(wait_future(f2));
+      continue;
+    }
+
+    int out_present;
+    char *val;
+    int vallen;
+    fdb_check(f1.get(&out_present, (const uint8_t **)&val, &vallen));
+
+    REQUIRE(out_present);
+    uint64_t transaction_id = std::stoul(std::string(val, vallen));
+    CHECK(transaction_id == ULONG_MAX);
+    break;
+  }
+}
+
+// TODO: Need to handle the exception generated by the write-after-write. Skip
+// this test for now...
+TEST_CASE("special-key-space set transaction ID after write"
+          * doctest::skip(true)) {
+  fdb::Transaction tr(db);
+  fdb_check(tr.set_option(FDB_TR_OPTION_SPECIAL_KEY_SPACE_ENABLE_WRITES,
+                          nullptr, 0));
+  while (1) {
+    tr.set(key("foo"), "bar");
+    tr.set("\xff\xff/tracing/a/transaction_id", "0");
```

Comment:
This `set` will cause an exception to be thrown in the special-key-space (this is the current implementation, but it can be changed: https://github.com/apple/foundationdb/pull/4125/commits/3669615e4bf9f5765f24366745f557bd872825d3#diff-74ee83047df93d7ae6bf63547f4e8b4b92742bb9f9d840539199f4967ee7a8bcR1293). The issue is, I'm not sure how to `catch` the error here, as it seems any exception that is thrown is handled above the client layer and causes a crash (meaning the unit test fails).

Reply:
oh, this is a problem I didn't think of... you could change the signature of the set so that it can return an error - though I am not sure whether that would be a good idea (it's a pretty big change for applications).

I think there are two possible solutions:

1. You ignore the set in this case (so don't throw in the special key space)
2. You remember that this happened and throw in commit.

I think both of the above approaches should be ok. Setting this key should not make a functional difference for the transaction, so we don't need to be too careful.